File: Computer Vision/reading images/readimg.m
close all, clear all, clc;

img = imread('seals.jpg');

[height, width, colors] = size(img)

# We have 3 channels on each pixel the width x height grid
# Each channel can store values from 0-255 (uint8)€
# These three channels represent Red, Green and Blue
# This approach for storing colors is called Trichotomy
figure(1), imshow(img);
title("Original image")

typeinfo(img)

# Extracting channels accordingly
red_channel = double(img(:, :, 1));
green_channel = double(img(:, :, 2));
blue_channel = double(img(:, :, 3));

# Creating a zeros channel with proper dimensions to zero out channels individually
zeros_channel = zeros(size(red_channel));

# Creating images with the extracted channels and zeros channel

# Red channel only image
img_red = uint8(cat(3, red_channel, zeros_channel, zeros_channel));
#figure(2), imshow(img_red), title("Red channel");

# Blue channel only image
img_blue = uint8(cat(3, zeros_channel, blue_channel, zeros_channel));
#figure(3), imshow(img_blue), title("Blue channel");

# Green channel only image
img_green = uint8(cat(3, zeros_channel, zeros_channel, green_channel));
#figure(4), imshow(img_green), title("Green channel");

# Red and green to represent yellow channel

img_yellow = uint8(cat(3, red_channel, green_channel, zeros_channel));
figure(5), imshow(img_yellow), title("Yellow channel");

# Image reconstruction using all channels again
img_reconstruct = uint8(cat(3,red_channel, green_channel, blue_channel));
#figure(6), imshow(img_reconstruct), title("Reconstructed image");

# Abstract images
img_abstract1 = uint8(cat(3,green_channel, blue_channel, red_channel));
#figure(7), imshow(img_abstract1), title("Abstract Image");

img_abstract2 = uint8(cat(3,blue_channel, red_channel, green_channel));
#figure(8), imshow(img_abstract2), title("Abstract Image");

File: Computer Vision/reading images/readimg.py
from PIL import Image
import numpy as np

image_path = "seals.jpg"
image = Image.open(image_path)

img = np.array(image)
width, height, channels = img.shape
print("width:",width, ", height:", height, ", channels:", channels, ", mode:", image.mode)

red = img[:,:,0]
green = img[:,:,1]
blue = img[:,:,2]
zero = np.zeros_like(red)

# image recreation
merged = np.stack([red, green, blue], axis=2)
merged_image = Image.fromarray(merged)
merged_image.show()

# red channel
red_image = np.stack([red, zero, zero], axis=2)
red_image = Image.fromarray(red_image)
#red_image.show()

# green channel
green_image = np.stack([zero, green, zero], axis=2)
green_image = Image.fromarray(green_image)
#green_image.show()

# blue channel
blue_image = np.stack([zero, zero, blue], axis=2)
blue_image = Image.fromarray(blue_image)
#blue_image.show()


# yellow channel
yellow_image = np.stack([red, green, zero], axis=2)
yellow_image = Image.fromarray(yellow_image)
#yellow_image.show()


# abstract images
abstract_image = np.stack([green, blue, red], axis=2)
abstract_image = Image.fromarray(abstract_image)
abstract_image.show()

# abstract images
abstract_image = np.stack([red*2, green*2, blue*2], axis=2)
abstract_image = Image.fromarray(abstract_image)
abstract_image.show()


